src/plots/scatterPlot.ts

```diff
@@ -205,4 +222,105 @@ export class Scatter<X, Y> extends XYPlot<X, Y> {
+    }
+  }
+
+  protected _additionalPaint(time: number) {
+    this.datasets().forEach((dataset) => this._labelConfig.get(dataset).labelArea.selectAll("g").remove());
+    if (this._labelsEnabled) {
+      Utils.Window.setTimeout(() => this._drawLabels(), time);
+    }
+  }
+
+  protected _drawLabels() {
+    const dataToDraw = this._getDataToDraw();
+    const attrToProjector = this._getAttrToProjector();
+    this.datasets().forEach((dataset) => {
+      dataToDraw.get(dataset).forEach((datum, index) => {
+        if (datum == null) {
+          return;
+        }
+        this._drawLabel(datum, index, dataset, attrToProjector);
+      });
+    });
+  }
+
+  private _drawLabel(datum: any, index: number, dataset: Dataset, attrToProjector: AttributeToProjector) {
+    if (datum.label == null) {
+      return;
+    }
+    const { labelArea, measurer, writer } = this._labelConfig.get(dataset);
+
+    const scatterCoordinates = { x: attrToProjector["x"](datum, index, dataset), y: attrToProjector["y"](datum, index, dataset) };
+
+    const sizeProjector = Plot._scaledAccessor(this.size());
+    const diameter = sizeProjector(datum, index, dataset);
+
+    const label = this._labelFormatter(datum.label, datum, index, dataset);
+    const measurement = measurer.measure(label);
+
+    const { containerDimensions, labelContainerOrigin, labelOrigin, alignment } = this._calculateLabelProperties(scatterCoordinates, diameter, measurement);
+
+    const labelContainer = this._createLabelContainer(labelArea, labelContainerOrigin, labelOrigin, measurement);
+
+    const writeOptions = { xAlign: alignment.x as Typesettable.IXAlign, yAlign: alignment.y as Typesettable.IYAlign };
+    writer.write(label, containerDimensions.width, containerDimensions.height, writeOptions, labelContainer.node());
+  }
+
+  private _calculateLabelProperties(
+    pointCoordinates: Point, diameter: number, measurement: Typesettable.IDimensions) {
+
+    // If diameter is smaller than font size, put label above
+
+    const labelShift = diameter < measurement.height ? diameter / 2 + Scatter._LABEL_MARGIN_FROM_BUBBLE : 0;
+
+    return {
+      containerDimensions: {
+        width: measurement.width,
+        height: measurement.height,
+      },
+      labelContainerOrigin: {
+        x: pointCoordinates.x - measurement.width / 2,
+        y: pointCoordinates.y - measurement.height / 2 + labelShift,
+      },
+      labelOrigin: {
+        x: pointCoordinates.x,
+        y: pointCoordinates.y,
+      },
+      alignment: {
+        x: "center",
+        y: "center",
+      },
+    };
+  }
+
+  private _createLabelContainer(
+    labelArea: SimpleSelection<void>, labelContainerOrigin: Point, labelOrigin: Point, measurement: Typesettable.IDimensions) {
+
+    const labelContainer = labelArea.append("g").attr("transform", `translate(${labelContainerOrigin.x}, ${labelContainerOrigin.y})`);
+    labelContainer.classed("on-bar-label", true);
+    labelContainer.classed("dark-label");
+
```

Comment:
Actually it looks like the second call is a no-op since it doesn't alter the label but rather returns a boolean indicating whether or not the selection has that class name.

https://github.com/d3/d3-selection/blob/master/README.md#selection_classed